pants

```diff
@@ -2,6 +2,32 @@
 # Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
+function expected_python_version_to_run() {
+  if [[ "$PANTS_USE_PYTHON3" == 'true' ]]; then
+    echo '3'
+  else
+    echo '2'
+  fi
+}
+
+_PYTHON_VERSION_FILE="./.pants-python-version"
+
+if [[ -z "$_RECURSIVE_PANTS_SHELL_SCRIPT_INVOCATION" ]]; then
+  _EXPECTED_PYTHON_VERSION="$(expected_python_version_to_run)"
+  _PREV_RUN_PYTHON_VERSION="$(cat 2>/dev/null "$_PYTHON_VERSION_FILE")"
+
+  if [[ "$_EXPECTED_PYTHON_VERSION" != "$_PREV_RUN_PYTHON_VERSION" ]]; then
+    export _RECURSIVE_PANTS_SHELL_SCRIPT_INVOCATION='y'
```

Comment:
Should this be `=true`? I've never seen someone use `y` in Bash to represent true (although I know very little bash).

Reply:
This is just intended to be "not matching `-z`" -- `true` is indeed more idiomatic and I will make that change.

Reply:
The alternative is `if [[ "$_RECURSIVE_PANTS_SHELL_SCRIPT_INVOCATION" != 'true' ]]`, which feels weird due to the negative. I'm not very opinionated about this at all though.

Reply:
The `y` isn't anything special otherwise.